### DISCUSSION TITLE
Yet another way of using filters

### DISCUSSION THREAD

**Discussion by brabebhin** [ffmpeginteropx/FFmpegInteropX#309] [Ideas]
Hi,

This is purely a selfish feature request, but here it goes.
My use case is built around gapless playback of video files (they are produced by a dashcam, and are really played gaplessly with MediaPlaybackList).

Buuut, I also want video effects, and as you know, that;s bugged.
So it occurred to me: We can use win2D directly in our decoding pipeline with both directx decoder and software decoder (no passthrough).

For DirectX path, the AVFrame is already a 2DTexture, and for software decoding we can create new ones based on the decoder buffer. This won't work with passthrough though. Then we can feed the 2DTexture into win2D. Then the win2D texture is passed through custom made effect graphs (we just expose an interface others can implement)

Would you agree to have such a feature, @lukasf ?

Thanks.

**Comment by lukasf** [comment]
Hi @brabebhin,

cool idea! Yeah I think this would be a clever way to work around the MediaPlaybackList filter limitations. I'm all for it. Maybe we could even get rid of the Win2D dependency, if we use use IDirect3DSurface on the interface as in and out parameter? And then have the effects filter as an additional library (with Win2D dependency ofc).

**Reply by brabebhin** [reply]
Good point about the win2D dependency. Getting rid of it from the main lib can be useful, especially in winui3 context.
